morpho-checkpoint-node/src/Morpho/Tracing/Pretty.hs

```diff
@@ -0,0 +1,397 @@
+        <+> viaShow epoch
+    ChainDB.OpenedVolatileDB -> "Opened vol db"
+    ChainDB.OpenedLgrDB -> "Opened lgr db"
+  mpretty (ChainDB.TraceFollowerEvent ev) = case ev of
+    ChainDB.NewFollower -> "New follower was created"
+    ChainDB.FollowerNoLongerInMem _ -> "FollowerNoLongerInMem"
+    ChainDB.FollowerSwitchToMem _ _ -> "FollowerSwitchToMem"
+    ChainDB.FollowerNewImmIterator _ _ -> "FollowerNewImmIterator"
+  mpretty (ChainDB.TraceInitChainSelEvent ev) = case ev of
+    ChainDB.InitChainSelValidation _ -> "InitChainSelValidation"
+  mpretty (ChainDB.TraceIteratorEvent ev) = case ev of
+    ChainDB.UnknownRangeRequested _ -> "UnknownRangeRequested"
+    ChainDB.BlockMissingFromVolatileDB _ -> "BlockMissingFromVolDB"
+    ChainDB.StreamFromImmutableDB _ _ -> "StreamFromImmDB"
+    ChainDB.StreamFromBoth {} -> "StreamFromBoth"
+    ChainDB.StreamFromVolatileDB {} -> "StreamFromVolDB"
+    ChainDB.BlockWasCopiedToImmutableDB _ -> "BlockWasCopiedToImmDB"
+    ChainDB.BlockGCedFromVolatileDB _ -> "BlockGCedFromVolDB"
+    ChainDB.SwitchBackToVolatileDB -> "SwitchBackToVolDB"
+  mpretty (ChainDB.TraceImmutableDBEvent ev) = "TraceImmutableDBEvent" <+> viaShow ev
+  mpretty (ChainDB.TraceVolatileDBEvent ev) = "TraceVolDBEvent" <+> viaShow ev
+
+instance
+  (Show (GenTxId blk), Show (ApplyTxErr blk), Show (GenTx blk)) =>
+  MPretty (TraceEventMempool blk)
+  where
+  mpretty = viaShow
+
+instance MPretty NtN.HandshakeTr where
+  mpretty = viaShow
+
+instance MPretty NtC.HandshakeTr where
+  mpretty = viaShow
+
+instance MPretty NtN.AcceptConnectionsPolicyTrace where
+  mpretty = viaShow
+
+instance MPretty DiffusionInitializationTracer where
+  mpretty = viaShow
+
+instance MPretty TraceLedgerPeers where
+  mpretty = viaShow
+
+instance (Show peer, Show a) => MPretty (TraceLabelPeer peer a) where
+  mpretty = viaShow
+
+instance MPretty (TraceTxSubmissionInbound txid tx) where
+  mpretty = viaShow
+
+instance (Show tx, Show txid) => MPretty (TraceTxSubmissionOutbound txid tx) where
+  mpretty = viaShow
+
+instance (Show addr) => MPretty (WithAddr addr ErrorPolicyTrace) where
+  mpretty = viaShow
+
+instance MPretty (WithDomainName (SubscriptionTrace Socket.SockAddr)) where
+  mpretty = viaShow
+
+instance MPretty (WithDomainName DnsTrace) where
+  mpretty = viaShow
+
+instance MPretty (WithIPList (SubscriptionTrace Socket.SockAddr)) where
+  mpretty = viaShow
+
+instance (Show peer) => MPretty (WithMuxBearer peer MuxTrace) where
+  mpretty = viaShow
+
+instance MPretty a => MPretty [a] where
+  mpretty xs = list $ map mpretty xs
+
+instance (StandardHash blk) => MPretty (ChainHash blk) where
+  mpretty = viaShow
+
+instance
+  (LedgerSupportsProtocol blk) =>
+  MPretty (ChainDB.InvalidBlockReason blk)
+  where
+  mpretty = viaShow
+
+condenseT :: Condense a => a -> Text
+condenseT = Text.pack . condense
+
+showPoint ::
+  Condense (HeaderHash blk) =>
+  TracingVerbosity ->
+  Point blk ->
+  String
+showPoint verb pt =
```

Comment:
This function is only called once with `verb = MaximalVerbosity`, so we might as well inline it.

Maybe even better, define an `instance Condense (HeaderHash blk) => MPretty (Point blk)` and use `mpretty` instead of `showPoint`